src/main/webapp/components/lists/list-create.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import FolderIcon from '@material-ui/icons/Folder'
+import ArchiveIcon from '@material-ui/icons/Archive'
+import CodeIcon from '@material-ui/icons/Code'
+import ListIcon from '@material-ui/icons/List'
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import ListItemText from '@material-ui/core/ListItemText'
+import FormControl from '@material-ui/core/FormControl'
+import TextField from '@material-ui/core/TextField'
+import MenuItem from '@material-ui/core/MenuItem'
+import Button from '@material-ui/core/Button'
+import { RadioButton } from '../input'
+import Popover from '@material-ui/core/Popover'
+import Box from '@material-ui/core/Box'
+import { v4 as newUuid } from 'uuid'
+
+const iconList = {
+  folder: <FolderIcon />,
+  code: <CodeIcon />,
+  archive: <ArchiveIcon />,
+  tasks: <ListIcon />,
+}
+
+const ListIconType = props => {
+  const { label } = props
+  return (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+      }}
+    >
+      <ListItemIcon>{iconList[label]}</ListItemIcon>
+      <ListItemText primary={label} />
+    </div>
+  )
+}
+
+export const ListCreatePopover = props => {
+  const { list, anchorEl, onClose, open, onSave } = props
+  return (
+    <Popover
+      open={open}
+      anchorEl={anchorEl}
+      onClose={onClose}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'left',
+      }}
+    >
+      <Box style={{ margin: '10px' }}>
+        <ListCreate prevList={list} onClose={onClose} onSave={onSave} />
+      </Box>
+    </Popover>
+  )
+}
+
+export const ListCreate = props => {
+  const { prevList, onClose, onSave } = props
+  const isNewList = prevList === undefined
+
+  const [list, setList] = React.useState({
+    id: isNewList ? newUuid().replace(/-/g, '') : prevList.id,
+    list_icon: isNewList ? 'folder' : prevList.list_icon,
+    list_bookmarks: isNewList ? [] : prevList.list_bookmarks,
+    query: isNewList ? undefined : prevList.query,
+    title: isNewList ? 'Untitled List' : prevList.title,
+  })
```

Comment:
This logic is kind of awkward. I think it would read better like this:
```suggestion
  const { onClose, onSave } = props
  const [list, setList] = React.useState(props.list || {
    id: newUuid().replace(/-/g, ''),
    list_icon: 'folder' ,
    list_bookmarks: [],
    query: undefined,
    title: 'Untitled List',
  })
```